helix-core/src/test/java/org/apache/helix/integration/TestPartitionMovementThrottle.java

```diff
@@ -211,8 +217,54 @@ public void testANYtypeThrottle() throws InterruptedException {
     Thread.sleep(2000L);
 
     for (int i = 0; i < NODE_NR; i++) {
-      validateThrottle(DelayedTransition.getInstancePatitionTransitionTimes(),
-          _participants[i].getInstanceName(), 1);
+      Assert.assertTrue(getMaxParallelTransitionCount(DelayedTransition.getInstancePatitionTransitionTimes(), _participants[i].getInstanceName()) <= 1);
+    }
+  }
+
+  @Test
+  public void testThrottleOnlyClusterLevelAnyType() {
+    // start some participants
+    for (int i = 0; i < NODE_NR - 3; i++) {
+      _participants[i].syncStart();
+    }
+    // Add resource: TestDB_ANY of 20 partitions
+    _gSetupTool.addResourceToCluster(CLUSTER_NAME, WorkflowGenerator.DEFAULT_TGT_DB + "_OnlyANY", 20,
+        STATE_MODEL, RebalanceMode.FULL_AUTO.name());
+    // Act the rebalance process
+    _gSetupTool.rebalanceStorageCluster(CLUSTER_NAME, WorkflowGenerator.DEFAULT_TGT_DB + "_OnlyANY",
+        _replica);
+
+    Assert.assertTrue(_clusterVerifier.verifyByPolling());
+
+    // overwrite the cluster level throttle configuration
+    ClusterConfig clusterConfig = _configAccessor.getClusterConfig(CLUSTER_NAME);
+    StateTransitionThrottleConfig anyTypeClusterThrottle = new StateTransitionThrottleConfig(
+        StateTransitionThrottleConfig.RebalanceType.ANY, StateTransitionThrottleConfig.ThrottleScope.CLUSTER, 1);
+    clusterConfig.setStateTransitionThrottleConfigs(ImmutableList.of(anyTypeClusterThrottle));
+    _configAccessor.setClusterConfig(CLUSTER_NAME, clusterConfig);
+
+    DelayedTransition.setDelay(20);
+    DelayedTransition.enableThrottleRecord();
+
+    List<MockParticipantManager> newNodes = Arrays.asList(_participants).subList(NODE_NR - 3, NODE_NR);
+    newNodes.forEach(MockParticipantManager::syncStart);
+    newNodes.forEach(node -> {
+      try {
+        Assert.assertTrue(TestHelper.verify(
+            () -> getMaxParallelTransitionCount(DelayedTransition.getInstancePatitionTransitionTimes(), node.getInstanceName()) <= 1, 1000 * 2));
+      } catch (Exception e) {
+        e.printStackTrace();
+        assert false;
+      }
+    });
+
+    ClusterLiveNodesVerifier clusterLiveNodesVerifier = new ClusterLiveNodesVerifier(
+        _gZkClient, CLUSTER_NAME, Lists.transform(Arrays.asList(_participants), MockParticipantManager::getInstanceName));
+    Assert.assertTrue(clusterLiveNodesVerifier.verifyByZkCallback(1000));
+    Assert.assertTrue(_clusterVerifier.verifyByPolling());
+
+    for (int i = 0; i < NODE_NR; i++) {
+      _participants[i].syncStop();
```

Comment:
Just be careful to check that _participants[i] != null and _participants[i].isConnected().
If you call syncStop() twice, I think the program hangs. So make sure all of these Participants are running before calling syncStop() on all of them.

Also please add a TestHelper.verify() with the check for LiveInstances to make sure that the number of LiveInstance ZNodes is 0.

Reply:
Sounds good. Added the verifier in the @aftermethod so every test needs to check if the cleanup work is finished.